contracts/LifMarketMaker.sol

```diff
@@ -97,22 +106,64 @@ contract LifMarketMaker is Ownable {
     ];
 
     uint256 accumDistribution = 0;
-    uint256 periodStartBlock = startBlock;
+    uint256 deltaDistribution = 0;
+    uint256 startBlockPeriod = startBlock;
 
     for (uint8 i = 0; i < totalPeriods; i++) {
-      uint256 deltaDistribution;
+
+      require(marketMakerPeriods.length <= i);
```

Comment:
Yes, the validation is to call this function only once, we can do it before the loop. I will do that.